openwisp_notifications/handlers.py

```diff
@@ -192,34 +223,32 @@ def send_email_notification(sender, instance, created, **kwargs):
     if not (email_preference and instance.recipient.email and email_verified):
         return
 
-    try:
-        subject = instance.email_subject
-    except NotificationRenderException:
-        # Do not send email if notification is malformed.
+    recipient_id = instance.recipient.id
+    cache_key = f'email_batch_{recipient_id}'
+    cache_key_pks = f'{instance.recipient.email}_pks'
+    cache_data = cache.get(
+        cache_key, {'last_email_sent_time': None, 'batch_scheduled': False}
+    )
+
+    if cache_data['last_email_sent_time'] and EMAIL_BATCH_INTERVAL > 0:
```

Comment:
It would be helpful to add some comments here explaining why this logic exists and describing the two cases it handles.